### PR TITLE
mrc-1235 final batch of translations

### PR DESCRIPTION
**Pull request by hillalex**
Components:
* barchart
* choropleth
* surveyAndProgram
* fileUpload
* resetConfirmation
* stepper/step

Contains commits from https://github.com/mrc-ide/hint/pull/223

### CODE REVIEW

File: src/app/static/src/app/components/plots/ChoroplethFilters.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
     <div v-if="hasFilters">
-        <h4>Filters</h4>
+        <h4 v-translate="'Filters'"></h4>
```
Comment (EmmaLRussell) [inline src/app/static/src/app/components/plots/ChoroplethFilters.vue]: I think the key is lower case, 'filters'..?

**Comment by hillalex** [comment]
> Nice one, this must have been boring!
> 
> So the only other rogue strings I can find in the front end are:
> /store/root/actions line 46 (loading data error)
> various error messages in apiService
> 
> Are these going into a separate branch?

Ah thanks, no I had missed those. I'll add them in here :+1:

File: src/app/static/src/app/apiService.ts [hunk 2]
```diff
@@ -46,15 +47,15 @@ export class APIService<S extends string, E extends string> implements API<S, E>
...
+            return APIService.createError("apiMissingError");
         }
         return failure.errors[0];
     };
 
     static createError(detail: string) {
         return {
             error: "MALFORMED_RESPONSE",
-            detail
+            detail: i18next.t(detail)
```
Comment (hillalex) [inline src/app/static/src/app/apiService.ts]: So this of course means that once the error is displayed, if the language is toggled it remains in the language it was generated with. But this is the same situation as with the errors coming back from the API.

Comment (EmmaLRussell) [reply src/app/static/src/app/apiService.ts]: I think that's fair enough!

**Pull request merged**